test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -1440,6 +1440,28 @@ static public void testDefaultValueInassortedRefWithLongAlignmentArray() throws
 		}
 	}
 
+	/*
+	 * Ensure that casting null to a value type class will throw a null pointer exception 
+	 */
+	@Test(priority=4, expectedExceptions=NullPointerException.class)
+	static public void testCheckCastValueTypeOnNull() throws Throwable {
+		String fields[] = {"longField:J"};
+		Class valueClass = ValueTypeGenerator.generateValueClass("TestCheckCastValueTypeOnNull", fields);
+		MethodHandle checkCastValueTypeOnNull = lookup.findStatic(valueClass, "testCheckCastValueTypeOnNull", MethodType.methodType(Object.class));
+		checkCastValueTypeOnNull.invoke();
+	}
+
+	/*
+	 * Ensure that casting null to a reference type class will pass
+	 */
+	@Test(priority=4)
```

Comment:
same here

Reply:
also add a test where a non-NULL VT is being casted

Reply:
For this test, is it sufficient to just use the DEFAULTVALUE instruction to create a value type and then just checkcast that with the same value type class used to create the object? Or would it be better to create a super class and a base class and try to cast to a super class value type object to a base type one?

Reply:
The super class of any VT is always j.l.Object so you can just checkcast to the same type.